.vitepress/theme/components/PreferenceTooltip.vue

```diff
@@ -0,0 +1,227 @@
+<script setup lang="ts">
+import { onMounted, ref, watch } from 'vue'
+import {
+  inBrowser,
+  preferComposition,
+  preferCompositionKey
+} from './preferences'
+import { useData, type Header } from 'vitepress'
+
+const show = ref(false)
+const { page } = useData()
+
+type Source = 'url-query' | 'url-header' | 'default'
+let source: Source | false =
+  inBrowser && localStorage.getItem(preferCompositionKey) === null
```

Comment:
Is this logic a bit too long in length for a ternery here? Think an if statement would make this cleaner?

Reply:
I think it reads more naturally with this ternary 🤔